drivers/ethernet/eth_adin2111.c

```diff
@@ -312,7 +312,7 @@ static void adin2111_offload_thread(const struct device *dev)
 		}
 
 		/* handle port 2 phy interrupts */
-		if (status1 & ADIN2111_STATUS1_PHYINT) {
+		if ((status1 & ADIN2111_STATUS1_PHYINT) && (adin_cfg->id == ADIN2111_MAC)) {
```

Comment:
One optimization improvement would be to create a bolean = (adin_cfg->id == ADIN2111_MAC) and then replace this statement in the file by the boolean. This would improve hardware efficiency and speed of execution. Obviously it depends on the compiler optimizations, etc but it also has the advantage of improving readaibility.